### ISSUE TITLE
HHH Benchmark evaluation question: why using base prompt and (A - A_base) > (B - B_base)?

### ISSUE THREAD

**Issue by t170815518** [declare-lab/instruct-eval#20]
Thanks for the insightful work! I have a question regarding the evaluation function in `hhh.py`, specifically [at line 101](https://github.com/declare-lab/instruct-eval/blob/e2eb966cdb395f65b5788a9c68f573e750e3b1f9/hhh.py#L101). Could you explain the purpose of the `base_prompt` here? Additionally, I'm curious about the rationale behind using the following code snippet:

```python
if (A - A_base) > (B - B_base):
    pred = "A"
    num_A += 1
else:
    pred = "B"
    num_B += 1
```

instead of the simpler alternative:
```python
if (A) > (B):
    pred = "A"
    num_A += 1
else:
    pred = "B"
    num_B += 1
```

Thank you for your response in advance!

**Comment by Emrys-Hong** [comment]
Hi Thanks for pointing out this issue. We have updated the code [here](https://github.com/declare-lab/instruct-eval/commit/fb247fff57a24c5b5e053ba25f0eac4835e1051f): 

We have tried the simpler alternative. However, We have found that the position of the options can affect the prediction (logits) by quite a lot.

This is similar to this paper: Large Language Models are not Fair Evaluators (https://arxiv.org/pdf/2305.17926.pdf). Where they report and I quote:
```
We find that the quality ranking of candidate responses can be
easily hacked by simply altering their order of appearance in the context
```

Therefore, we need to consider the logit of the correct answer when it is at different positions for LLM to give more robust and fair answers.

I have made some comments on the code for you to understand.

Please let me know if you need more explanation, if not please kindly to close the issue.
